File: backend-app/src/main/java/com/gradientgeeks/ageis/backendapp/dto/SignatureValidationResponse.java
package com.gradientgeeks.ageis.backendapp.dto;

/**
 * DTO for signature validation response from Aegis API.
 */
public class SignatureValidationResponse {
    
    private boolean isValid;
    private String message;
    private String deviceId;
    
    // Constructors
    public SignatureValidationResponse() {
    }
    
    public SignatureValidationResponse(boolean isValid, String message, String deviceId) {
        this.isValid = isValid;
        this.message = message;
        this.deviceId = deviceId;
    }
    
    // Getters and Setters
    public boolean isValid() {
        return isValid;
    }
    
    public void setValid(boolean valid) {
        isValid = valid;
    }
    
    public String getMessage() {
        return message;
    }
    
    public void setMessage(String message) {
        this.message = message;
    }
    
    public String getDeviceId() {
        return deviceId;
    }
    
    public void setDeviceId(String deviceId) {
        this.deviceId = deviceId;
    }
}

File: sfe/sfe-client/src/main/java/com/gradientgeeks/aegis/sfe_client/model/SignatureValidationResponse.kt
package com.gradientgeeks.aegis.sfe_client.model

import com.google.gson.annotations.SerializedName

/**
 * Response model for signature validation from the Aegis Security API.
 * 
 * Contains the result of signature verification performed by the backend.
 */
data class SignatureValidationResponse(
    
    /**
     * Boolean indicating whether the signature is valid.
     * True if the signature was successfully verified, false otherwise.
     */
    @SerializedName("isValid")
    val isValid: Boolean,
    
    /**
     * Human-readable message describing the validation result.
     * May contain additional details about validation failure reasons.
     */
    @SerializedName("message")
    val message: String,
    
    /**
     * Echo of the device ID that was validated.
     * Useful for request correlation and debugging.
     */
    @SerializedName("deviceId")
    val deviceId: String? = null
)